youtube_dl/extractor/youmaker.py

```diff
@@ -0,0 +1,196 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..compat import compat_urllib_parse_urlencode
+from ..utils import parse_iso8601, unified_strdate, ExtractorError
+
+
+class YouMakerIE(InfoExtractor):
+    _VALID_URL = r"""(?x)
+                    https?://(?:www\.)?youmaker\.com/
+                    (?:v|video|embed)/
+                    (?P<id>[0-9a-zA-Z-]+)
+                    """
+
+    _TESTS = [
+        {
+            "url": "http://www.youmaker.com/v/71b5d2c5-31b6-43b8-8475-1dcb5e10dfb0",
+            "info_dict": {
+                "id": "71b5d2c5-31b6-43b8-8475-1dcb5e10dfb0",
+                "ext": "mp4",
+                "title": "Как сшить шапочку из трикотажа. Плоский шов двойной иглой.",
+                "description": r"re:(?s)^Привет друзья!\n\nВ этом видео я .* представлена www\.iksonmusic\.com$",
+                "thumbnail": r"re:^https?://.*\.(?:jpg|png)$",
+                "duration": 358,
+                "upload_date": "20190614",
+                "uploader": "user_318f21e00e1f8a6b414f20a654d0f4fc7d2053bc",
+                "timestamp": 1560554895,
+                "channel": "Sewing Ideas",
+                "channel_id": "40ca79f7-8b21-477f-adba-7d0f81e5b5fd",
+                "channel_url": r"re:https?://www.youmaker.com/channel/40ca79f7-8b21-477f-adba-7d0f81e5b5fd",
+                "tags": [
+                    "как сшить детскую шапочку из трикотажа",
+                    "как шить двойной иглой трикотаж",
+                ],
+                "categories": ["Life", "How-to & DIY"],
+            },
+            "params": {
+                "skip_download": True,
+            },
+        },
+    ]
+
+    def __init__(self, downloader=None):
+        """Constructor. Receives an optional downloader."""
+        super(YouMakerIE, self).__init__(downloader=downloader)
+        self.__protocol = "https"
+        self.__video_id = None
+        self.__cache = {}
+
+    def _set_id_and_protocol(self, url):
+        self.__video_id = self._match_id(url)
+        if url.startswith("http://"):
+            self.__protocol = "http"
+
+    def _fix_url(self, url):
+        if url.startswith("//"):
+            return "%s:%s" % (self.__protocol, url)
+        return url
+
+    @property
+    def _base_url(self):
+        return self._fix_url("//www.youmaker.com")
+
+    @property
+    def _asset_url(self):
+        # as this url might change in the future
+        # it needs to be extracted from some js magic...
+        return self._fix_url("//vs.youmaker.com/assets")
+
+    def _api(self, path, cache=False, **kwargs):
```

Comment:
Some 62 out of 64 other extractors that do a similar thing have called the corresponding method `_call_api()`. I'm only pointing this out in case you might want to do so.

Reply:
ok
consistency is king and tool-based refactoring is easy :)